### PR TITLE
Fix scroll behavior of graph canvas.

### PR DESCRIPTION
**Pull request by rolandpoulter**
Before this fix scrolling on from an apple trackpad was hard to controller. Now it will behavior more like it would when using a mouse scroll wheel.

### CODE REVIEW

**Comment by heckj** [comment]
+1

**Comment by benbp** [comment]
:+1:

File: apps/graph_canvas/views/Viewport.js [hunk 1]
```diff
@@ -177,19 +177,38 @@ export default class GCViewport extends Component {
...
+    if (event.preventDefault) { event.preventDefault(); }
+    event.timeStamp = event.timeStamp || Date.now();
+    if (!this.scrollBuffer) {
+      this.scrollBuffer = [];
+      this.scrollBuffer.timeStamp = event.timeStamp + 100;
+    }
+    if (this.scrollBuffer.wait) { return; }
+    if (event.deltaY) { this.scrollBuffer.push(event.deltaY); }
+    clearTimeout(this.scrollBuffer.timer);
+    if (this.scrollBuffer.timeStamp <= event.timeStamp) {
```
Comment (rolandpoulter) [inline apps/graph_canvas/views/Viewport.js]: Sample `event.deltaY` buffer every tenth of a second.

File: apps/graph_canvas/views/Viewport.js [hunk 2]
```diff
@@ -177,19 +177,38 @@ export default class GCViewport extends Component {
...
+      let scale = this.graphCanvas.scale,
+          factor = 1.5,
+          force = this.scrollBuffer.length && this.scrollBuffer.reduce((a, b) => a + b);
+      scale = Math.max(0.0001, Math.min(100,
+        Math.abs(force < 0 ? scale / factor : scale * factor)
```
Comment (rolandpoulter) [inline apps/graph_canvas/views/Viewport.js]: Determines which direction the user wants to scroll and increase/decrease `scale` by a `factor` of `1.5`

File: apps/graph_canvas/views/Viewport.js [hunk 3]
```diff
@@ -177,19 +177,38 @@ export default class GCViewport extends Component {
...
+        scale = this.graphCanvas.scale;
+      }
+      this.scrollBuffer.wait = true;
+      this.graphCanvas.updateScale(Math.abs(scale), () => {
+        this.scrollBuffer = null;
+      });
     }
     else {
-      scale = Math.min(8, scale + force);
+      this.scrollBuffer.timer = setTimeout(
```
Comment (rolandpoulter) [inline apps/graph_canvas/views/Viewport.js]: Ensure the current `scrollBuffer` is resolved even if the scroll events stop.

**Pull request merged**